src/tests/integration/python/test_python.py

```diff
@@ -92,3 +95,48 @@ def test__python_project_install_lint_and_publish(
     kraken_ctx.load_project(directory=tempdir / consumer_dir)
     kraken_ctx.execute([":python.install"])
     # TODO (@NiklasRosenstein): Test importing the consumer project.
+
+
+@unittest.mock.patch.dict(os.environ, {})
+def test__python_project_upgrade_python_version_string(
+    kraken_ctx: Context,
+    tempdir: Path,
+) -> None:
+    project_dir = "version-project"
+    # FIXME - now this is hardcoded also in .kraken.py file, find a way to pass it there from test level
+    build_as_version = "9.9.9a1"
+    init_file_dir = "src/version_project/__init__.py"
+    project_dist = tempdir / project_dir / "build/python-dist"
+
+    # Copy the projects to the temporary directory.
+    shutil.copytree(Path(__file__).parent / "data" / project_dir, tempdir / project_dir)
```

Comment:
suggestion:

```suggestion
    shutil.copytree(original_dir, tempdir)
```

There's multiple copies of `Path(__file__).parent / "data" / project_dir` which would could just assign to a variable. Opting for `original_dir` here because `src_dir` could be confusing and sound like it's referring to a directory containing source code files.

You don't need to create a subdirectory inside the tempdir, so just using `tempdir` as the destination for the copy of the integration test directory is fine.

Reply:
I replaced path with `original_dir`, 

for the `tempdir` I keep creating the additional folder layer - this way I can use same copy function as in other tests (`shutil.copytree` - which in python 3.7 and older does not support copy to existing folder, that is `dirs_exist_ok` flag)